code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -361,97 +314,192 @@
 		mb.fail(src)
 		mb.forceMove(get_turf(src))
 	for(var/obj/O in contents)
-		amount++
-		if(O.reagents)	//this is reagents in inserted objects (like chems in produce)
-			var/id = O.reagents.get_master_reagent_id()
-			if(id)
-				amount+=O.reagents.get_reagent_amount(id)
+		if(O.reagents?.total_volume)
+			amount += O.reagents.total_volume
 		qdel(O)
-	if(reagents && reagents.total_volume)	//this is directly-added reagents (like water added directly into the machine)
-		var/id = reagents.get_master_reagent_id()
-		if(id)
-			amount += reagents.get_reagent_amount(id)
+	if(reagents?.total_volume)
+		amount += reagents.total_volume
 	reagents.clear_reagents()
 	if(amount)
-		var/obj/item/reagent_containers/food/snacks/badrecipe/ffuu = new(src)
-		ffuu.reagents.add_reagent("carbon", amount)
-		ffuu.reagents.add_reagent("????", amount/10)
-		ffuu.forceMove(get_turf(src))
+		var/obj/item/reagent_containers/food/snacks/badrecipe/mysteryfood = new(src)
+		mysteryfood.reagents.add_reagent("carbon", amount/2)
+		mysteryfood.reagents.add_reagent("????", amount/15)
+		mysteryfood.forceMove(get_turf(src))
 
-/obj/machinery/kitchen_machine/Topic(href, href_list)
-	if(..() || panel_open)
+/obj/machinery/kitchen_machine/update_icon_state()
+	. = ..()
+	if(dirty == 100)
+		icon_state = dirty_icon
 		return
-
-	usr.set_machine(src)
 	if(operating)
-		updateUsrDialog()
-		return
-
-	switch(href_list["action"])
-		if("cook")
-			cook()
-
-		if("dispose")
-			dispose()
-	return
+		icon_state = on_icon
+	else
+		icon_state = off_icon
 
 /obj/machinery/kitchen_machine/build_reagent_description(mob/user)
 	return
 
-/obj/machinery/kitchen_machine/examine(mob/user, infix = "", suffix = "")
+/obj/machinery/kitchen_machine/examine(mob/user)
 	. = ..()
 	var/dat = format_content_descs()
 	if(dat)
-		. += "It contains: <BR>[dat]"
+		. += "It contains: <br>[dat]"
+
+/obj/machinery/kitchen_machine/attack_hand(mob/user)
+	if(stat & (BROKEN|NOPOWER) || panel_open || !anchored)
+		return
+
+	ui_interact(user)
+
+/obj/machinery/kitchen_machine/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "KitchenMachine",  name, 400, 300, master_ui, state)
+		ui.open()
+
+/obj/machinery/kitchen_machine/ui_data(mob/user)
+	var/list/data = list()
+	data["operating"] = operating
+	data["inactive"] = FALSE
+	data["no_eject"] = FALSE
+	data["tooltip"] = ""
+	if(dirty >= 100)
+		data["inactive"] = TRUE
+		data["tooltip"] = "It's too dirty."
+	else if(!has_cookables())
+		data["inactive"] = TRUE
+		data["no_eject"] = TRUE
+		data["tooltip"] = "There are no contents."
+
+	var/list/items_counts = list()
+	var/list/name_overrides = list()
+	for(var/obj/O in contents)
+		var/display_name = O.name
+		if(istype(O,/obj/item/reagent_containers/food/snacks/egg)) // God I fuckin hate this
```

Comment:
I agree with your comments. The best solution to this would be to create a "name override" or "ingredient name" variable on the food type so that you can just access them here. You wouldn't have to cherry pick istype then.

Reply:
Implemented a version of this in the most recent commit, feedback on the new system would be appreciated